src/link/database.rs

```diff
@@ -1,33 +1,41 @@
+use apollo_compiler::ast::{Directive, DirectiveLocation};
+use apollo_compiler::schema::DirectiveDefinition;
+use apollo_compiler::{ty, Schema};
+use std::borrow::Cow;
 use std::{collections::HashMap, sync::Arc};
 
-use apollo_compiler::ast::{Directive, DirectiveLocation, Type};
-use apollo_compiler::Schema;
-
 use crate::link::{
     spec::{Identity, Url},
     {Link, LinkError, LinksMetadata, DEFAULT_LINK_NAME},
 };
 
+/// Extract @link metadata from a schema.
 pub fn links_metadata(schema: &Schema) -> Result<Option<LinksMetadata>, LinkError> {
     let mut bootstrap_directives = schema
         .schema_definition
         .directives
         .iter()
-        .filter(|d| parse_link_if_bootstrap_directive(schema, d));
-    let bootstrap_directive = bootstrap_directives.next();
-    if bootstrap_directive.is_none() {
+        .filter(|d| is_bootstrap_directive(schema, d));
+    let Some(bootstrap_directive) = bootstrap_directives.next() else {
         return Ok(None);
-    }
-    // We have _a_ bootstrap directives, but 2 is more than we bargained for.
-    if bootstrap_directives.next().is_some() {
+    };
+    // There must be exactly one bootstrap directive.
+    if let Some(extraneous_directive) = bootstrap_directives.next() {
         return Err(LinkError::BootstrapError(format!(
             "the @link specification itself (\"{}\") is applied multiple times",
-            Identity::link_identity()
+            extraneous_directive
+                .argument_by_name("url")
+                // XXX(@goto-bus-stop): @core compatibility is primarily to support old tests--should be
+                // removed when those are updated.
+                .or(extraneous_directive.argument_by_name("feature"))
+                .and_then(|value| value.as_str().map(Cow::Borrowed))
+                .unwrap_or_else(|| Cow::Owned(Identity::link_identity().to_string()))
         )));
     }
+
     // At this point, we know this schema uses "our" @link. So we now "just" want to validate
     // all of the @link usages (starting with the bootstrapping one) and extract their metadata.
```

Comment:
https://specs.apollo.dev/link/v1.0/ does not have a corresponding section but https://specs.apollo.dev/core/v0.2/#@core/feature has an extra check that the boostrap directive should be the first one listed with that name.

Reply:
Yeah it's a good point. JS doesn't check this, I would be surprised if there are any schemas where it's not true (since supergraphs are virtually always generated) but I don't really want to risk that incompatibility. I added a comment about it at the top of the function where we look up the bootstrap directive.